packages/pwa-kit-react-sdk/src/ssr/universal/contexts/index.js

```diff
@@ -5,9 +5,43 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import React from 'react'
+import React, {useEffect, useRef} from 'react'
+import PropTypes from 'prop-types'
+import {useLocation} from 'react-router-dom'
+import {uuidv4} from 'pwa-kit-runtime/utils/uuidv4'
+
 import DeviceContext from '../device-context'
 
 const ExpressContext = React.createContext()
 
-export {DeviceContext, ExpressContext}
+const CorrelationIdContext = React.createContext()
+
+const CorrelationIdProvider = ({children, correlationId}) => {
+    const [id, setId] = React.useState(correlationId || uuidv4())
+    const location = useLocation()
+
+    const isFirstRun = useRef(true)
+
+    useEffect(() => {
+        // don't run this on first render
+        if (isFirstRun.current) {
+            isFirstRun.current = false
+            return
+        }
+        const newId = uuidv4()
+        setId(newId)
+    }, [location.pathname])
```

Comment:
Some people have pages that vary on GET params – you've defined a page as a path only. How are ya gonna handle that?

Reply:
Correct me if I'm wrong but `react-router` doesn't take into account params when routing unless you were to do some magic things to route within your own page, and I believe that would go beyond what our definition of a page is.